src/Service/ProducerEventProcessorTest.cpp

```diff
@@ -1644,6 +1677,309 @@ TEST(ProducerEventProcessor, MemoryUsageEvent) {
   ASSERT_EQ(actual_memory_usage_event.SerializeAsString(), memory_usage_event->SerializeAsString());
 }
 
+TEST(ProducerEventProcessor, ApiScopeStart) {
+  ProducerCaptureEvent producer_capture_event;
+  ApiScopeStart* api_scope_start = producer_capture_event.mutable_api_scope_start();
+  api_scope_start->set_pid(kPid1);
+  api_scope_start->set_tid(kTid1);
+  api_scope_start->set_timestamp_ns(kTimestampNs1);
+  api_scope_start->set_color_rgba(kColor1);
+  api_scope_start->set_group_id(kGroupId1);
+  api_scope_start->set_address_in_function(kFunctionId1);
+  api_scope_start->set_encoded_name_1(kEncodedName1);
+  api_scope_start->set_encoded_name_2(kEncodedName2);
+  api_scope_start->set_encoded_name_3(kEncodedName3);
+  api_scope_start->set_encoded_name_4(kEncodedName4);
+  api_scope_start->set_encoded_name_5(kEncodedName5);
+  api_scope_start->set_encoded_name_6(kEncodedName6);
+  api_scope_start->set_encoded_name_7(kEncodedName7);
+  api_scope_start->set_encoded_name_8(kEncodedName8);
+  api_scope_start->mutable_encoded_name_additional()->push_back(kEncodedNameAdditional1);
```

Comment:
I kind of feel like it's already time to write a utility function to write to and to read encoded strings. Then here you would simply set the string, retrieve it afterwards, and compare, instead of comparing the individual fields. What do you think?

Reply:
I was thinking about this. My thought was, that this test actually does not care about the meaning of the data. It only ensures that the fields of the "converted" proto contains the unmodified values. So, I'd argue to keep it for now.

Reply:
Yeah, I know, it was more about making the code a bit simpler. Speaking of which, what do you think of using `MessageDifferencer` as pointed out in the chat in my question from yesterday? This is a lot of mechanical code and adding or removing fields would get lost here.

Reply:
Of course that would need to make a copy of the original proto

Reply:
Done.